src/software/ai/hl/stp/play/kickoff_enemy_play_test.cpp

```diff
@@ -28,18 +32,59 @@ TEST_F(KickoffEnemyPlayTest, test_kickoff_enemy_play)
     setRefereeCommand(RefereeCommand::NORMAL_START, RefereeCommand::PREPARE_KICKOFF_THEM);
 
     std::vector<ValidationFunction> terminating_validation_functions = {
-        // This will keep the test running for 9.5 seconds to give everything enough
-        // time to settle into position and be observed with the FullSystemGUI
-        // TODO: Implement proper validation
-        // https://github.com/UBC-Thunderbots/Software/issues/1397
         [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
-            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
+            // TODO: Fix bug with robot three not shadowing the enemy kicker in
+            // kickoff_enemy_play
+            // https://github.com/UBC-Thunderbots/Software/issues/1945
+
+            // Friendly robots in position to shadow enemy robots
+            Rectangle robotOneShadowingRect(Point(0, 2.2), Point(-0.4, 1.8));
+            Rectangle robotFiveShadowingRect(Point(0, -2.2), Point(-0.4, -1.8));
+            // Rectangle robotThreeShadowingRect(Point(-0.49, 0.1), Point(-0.75,
+            // -0.1));
+
+            robotInPolygon(1, robotOneShadowingRect, world_ptr, yield);
+            robotInPolygon(5, robotFiveShadowingRect, world_ptr, yield);
+            // robotInPolygon(3, robotThreeShadowingRect, world_ptr);
+
+            auto robotsDefendingPosts = [](std::shared_ptr<World> world_ptr) -> bool {
+                // Positions taken from kickoff_enemy_play
+                Point robotTwoExpectedPos =
```

Comment:
shouldn't our tests be non-implementation dependent @jonathanlew @akhilveeraghanta?
Maybe we can generalize this a bit more?

Reply:
We could generalize it to check if two robots are "near" the friendly defense area, in which case we can switch to the crease defender tactic and this test will still pass. 

Having robots near the defense area is pretty much all we care about here, their exact locations can change a bit

Reply:
I've updated the test to check that the two defending robots are generally positioned outside of the defense box.